packages/pigeon/lib/swift_generator.dart

```diff
@@ -637,11 +637,26 @@ import FlutterMacOS
     });
   }
 
+  void _writeNilOrType(Indent indent) {
+    indent.format('''
+
+private func nilOrValue<T>(value: Any?) -> T? {
+  if value is NSNull {
+    return nil
+  }
+  if let convertedValue = value as? T {
```

Comment:
It's not clear to me what this block is doing that's not redundant with the last `return`. If we need this, can you add a Dart comment above the `indent.format` statement explaining it?